coalib/settings/ConfigurationGathering.py

```diff
@@ -179,13 +182,16 @@ def warn_nonexistent_targets(targets, sections, log_printer=None):
 
     # Can't be summarized as python will evaluate conditions lazily, those
     # functions have intended side effects though.
-    files_config_absent = warn_config_absent(sections, 'files')
-    bears_config_absent = warn_config_absent(sections, ['bears', 'aspects'])
-    if files_config_absent or bears_config_absent:
+    files_config_absent = warn_config_absent(sections, 'files',
+                                             only_global_bears)
+    bears_config_absent = warn_config_absent(sections, ['bears', 'aspects'],
+                                             only_global_bears)
+    if (files_config_absent or bears_config_absent) and (
+            not (only_global_bears and files_config_absent)):
```

Comment:
Could you explain me this addition? Just curious what it does^^ Can't get behind it^^

Reply:
Yes the first part of the condition i.e. what was already there before deals with, if either the files parameter or bears parameter is missing, it will raise system exit whereas what i have added is `and (not (only_global_bears and files_config_absent)):` not to raise system exit in case of of only global bears are absent and files parameter is absent

Reply:
I think here's a mistake. I tried to reduce the expression, and I got:
```python
(file_config_absent and not only_global_bears) or
(bears_config_absent and not files_config_absent)
```
And the 2nd term is a bit wondering me: If both are True, it depends on `only_global_bears`. Shouldn't we always log something if we don't give any bears at all?

Reply:
@Makman2 if both are true, `only_global_bears` would be False, so an error would be logged

Reply:
and if files is given, and bears are not given, again, `only_global_bears` would be False and hence error would be logged

Reply:
all this due to added lines 502-509

Reply:
which are now on line 503 & 504